scripts/posteffects/posteffect-ssao.js

```diff
@@ -0,0 +1,568 @@
+// The implementation is based on the code in Filament Engine: https://github.com/google/filament
+// specifically, shaders here: https://github.com/google/filament/tree/24b88219fa6148b8004f230b377f163e6f184d65/filament/src/materials/ssao
+
+// --------------- POST EFFECT DEFINITION --------------- //
+/**
+ * @class
+ * @name SSAOEffect
+ * @classdesc Implements the SSAOEffect post processing effect.
+ * @description Creates new instance of the post effect.
+ * @augments PostEffect
+ * @param {GraphicsDevice} graphicsDevice - The graphics device of the application.
+ * @param {any}  ssaoScript - The script using the effect.
+ */
+function SSAOEffect(graphicsDevice, ssaoScript) {
+    pc.PostEffect.call(this, graphicsDevice);
+
+    this.ssaoScript = ssaoScript;
+    this.needsDepthBuffer = true;
+
+    this.ssaoShader = new pc.Shader(graphicsDevice, {
+        attributes: {
+            aPosition: pc.SEMANTIC_POSITION
+        },
+        vshader: [
+            (graphicsDevice.webgl2) ? ("#version 300 es\n\n" + pc.shaderChunks.gles3VS) : "",
```

Comment:
Are we sure this line is needed? The other post effects done declare this.

Reply:
yes those are neeed to use derivatives in webgl2. Chunk based shaders handle this automatically, but manual shaders don't.
But while looking at this, I got SSAO to work on webgl1 as well .. so worth spending time on this!